uwblueprint/supportive-housing
# Require 2FA On Initial Signup for Relief Staff

## Notion task link

2FA On Initial Signup (https://www.notion.so/uwblueprintexecs/2FA-On-Initial-Signup-f681347ebac54fb0b8bb6fa75c5a34f7)



## Implementation description
* Backend: modify /register route to also return whether the user needs 2FA or not (i.e. Relief Staff or not)
* Frontend: toggle from Signup page to 2FA page upon signup for Relief Staff; redirect to Home page for other users



## Steps to test
1. Login as an Admin and invite 3 new users: (a) an Admin, (b) a Relief staff who requires 2FA (check off this box when creating this user), and (c) a Relief Staff who does not require 2FA (don't check off the box).
2. From the Login page, click "Sign Up" and sign up as each of these 3 users. 
3. For (a) and (c), after signing up, you should be redirected to the Home page immediately. For (b), you should be prompted to complete 2FA with Authy.



## What should reviewers focus on?


## Checklist
